crates/tako/src/server/monitoring.rs

```diff
@@ -0,0 +1,73 @@
+use crate::messages::gateway::LostWorkerReason;
+use crate::messages::worker::WorkerOverview;
+use crate::server::core::CoreRef;
+use crate::WorkerId;
+
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MonitoringEvent {
+    pub message: MonitoringEventsPayload,
+    pub event_id: u32,
+    pub event_time: SystemTime,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum MonitoringEventsPayload {
+    WorkerConnected(WorkerId),
+    WorkerLost(WorkerId, LostWorkerReason),
+    OverviewReceived(WorkerId, WorkerOverview),
+}
+
+pub struct ServerStateMonitor {
+    core_ref: CoreRef,
+}
+
+impl ServerStateMonitor {
```

Comment:
There's a circular dependency between `Core` and the events, let's break it.
This struct should serve to:
1) Hold the events in memory.
2) Provide helper methods for various events that can be monitored.
It shouldn't know anything about `Core`.